syne_tune/optimizer/schedulers/hyperband.py

```diff
@@ -978,6 +1020,19 @@ def on_trial_complete(self, trial: Trial, result: Dict[str, Any]):
         self.searcher.cleanup_pending(trial_id)
         self._cleanup_trial(trial_id, trial_decision=SchedulerDecision.STOP)
 
+    def callback_for_checkpoint_removal(
```

Comment:
As by your suggestion, the callback is created and returned here, instead of outsourcing this to a factory.